torch/csrc/jit/jit_log.cpp

```diff
@@ -0,0 +1,98 @@
+#include <torch/csrc/jit/jit_log.h>
+#include <c10/util/Exception.h>
+#include <caffe2/utils/string_utils.h>
+#include <cstdlib>
+#include <iomanip>
+#include <mutex>
+#include <regex>
+#include <sstream>
+#include <thread>
+
+namespace torch {
+namespace jit {
+
+JitLoggingLevels jit_log_level() {
+  static const char* c_log_level = std::getenv("PYTORCH_JIT_LOG_LEVEL");
+  static const JitLoggingLevels log_level = c_log_level
+      ? static_cast<JitLoggingLevels>(std::atoi(c_log_level))
+      : JitLoggingLevels::OFF;
+  return log_level;
+}
+
+static std::string level_and_tid(JitLoggingLevels level) {
+  std::stringstream ss;
+  ss << "[" << level << " T";
+  ss << std::this_thread::get_id();
+  ss << "]: ";
+  return ss.str();
+}
+
+std::once_flag file_filter_initialized;
+std::vector<std::regex> file_regexes;
+
+bool apply_file_regex(const char* filename) {
+  static const char* c_file_filter = getenv("PYTORCH_JIT_LOG_FILE_FILTER");
+  if (!c_file_filter) {
+    return true;
+  }
+
+  std::call_once(file_filter_initialized, [&]() {
+    const auto str_regexes = caffe2::split(':', c_file_filter);
+
+    for (const auto& str_regex : str_regexes) {
+      file_regexes.emplace_back(std::regex(str_regex));
```

Comment:
std::regex is seriously broken on platforms like windows. I would not recommend adding a dependency on it.

Reply:
I removed this logic since you are against extra options